spec/features/saml/saml_spec.rb

```diff
@@ -167,6 +167,18 @@
   end
 
   context 'visiting /api/saml/logout' do
+    context 'via SP-initiated logout when not logged in to IdP' do
+      it 'does not raise an exception' do
+        request = OneLogin::RubySaml::Logoutrequest.new
+        settings = sp1_saml_settings
+        settings.name_identifier_value = SecureRandom.uuid
+
+        expect do
+          visit request.create(settings)
+        end.to_not raise_error
+      end
+    end
```

Comment:
> I don't see a functional difference between "log in and then timeout" and "never log in" -- what exactly is the difference?

Log out is a function of a logged in user; whether they have an active session at the IdP or not. "never log in is" an edge case that we're not discussing right now; and the server should respond with an error to that scenario.